src/pypdfium2/_library_scope.py

```diff
@@ -35,7 +84,42 @@ def destroy_lib():  # pragma: no cover
     pdfium_i.LIBRARY_AVAILABLE.value = False
 
 
-# Load pdfium
+def initialize_with_fonts(font_paths: Optional[List[str]] = None):
+    """
+    Initialize pypdfium2 with custom font paths.
+
+    This function should be called before any other pypdfium2 operations if you want to use
+    custom local fonts instead of or in addition to system fonts.
+
+    Args:
+        font_paths: List of paths to font directories or specific font files (TTF, OTF, TTC).
+                   PDFium will use these fonts when rendering PDFs.
+
+    Example:
+        import pypdfium2
+
+        # Initialize with custom fonts
+        pypdfium2.initialize_with_fonts([
+            "/path/to/fonts/folder",
+            "/path/to/specific/font.ttf"
+        ])
+
+        # Now use pypdfium2 normally
+        pdf = pypdfium2.PdfDocument("document.pdf")
+        ...
+    """
+    # If library is already initialized, we need to destroy and reinitialize
+    if pdfium_i.LIBRARY_AVAILABLE:
+        if pdfium_i.DEBUG_AUTOCLOSE:  # pragma: no cover
+            pdfium_i._safe_debug("Re-initializing PDFium with custom fonts")
+        # Note: This is not ideal but necessary for the API design
+        destroy_lib()
+        init_lib(font_paths)
```

Comment:
This re-initialization approach is also a no-go with me. See https://github.com/pypdfium2-team/pypdfium2/pull/380#issuecomment-3366972733 for what a proper alternative.